### PR TITLE
Add template Java dockerfile and automate build

### PR DESCRIPTION
**Pull request by leobr29**

### CODE REVIEW

**Comment by MCazaly** [comment]
Should we merge https://github.com/British-Oceanographic-Data-Centre/amrit-repos/pull/5 before reviewing this?

**Comment by leobr29** [comment]
> Should we merge #5 before reviewing this?

Yes, it's done !

File: java-demo/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+COPY ./.mvn ./.mvn
+COPY ./mvnw ./mvnw
+COPY ./pom.xml ./pom.xml
+COPY ./src ./src
+
+RUN ./mvnw clean package
+
+FROM eclipse-temurin:21-jre-alpine AS runtime
+
```
[the remark this answers is not in the document]
Comment (leobr29) [reply java-demo/Dockerfile:13]: I wasn't familiar with this, but great idea ! [603c5ee](https://github.com/British-Oceanographic-Data-Centre/amrit-repos/pull/6/commits/603c5ee589a4bf2a774ea81c60a727da19edc448)

**Comment by leobr29** [comment]
@MCazaly also added some tools and docs to run the tagged image from the registry (not simple as it requires a database). Actually you need to clone the repository to get docker compose files, I'll certainly have to improve that in the future.

File: java-demo/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+# Common base image.
+LABEL org.opencontainers.image.authors="matcaz@noc.ac.uk"
```
Comment (MCazaly) [inline java-demo/Dockerfile:15]: I don't think I should be listed as the author for this one! 😄 
```suggestion
LABEL org.opencontainers.image.authors="Leo.Bruvry.Lagadec@ifremer.fr"
```

File: java-demo/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+LABEL org.opencontainers.image.description="Example OCI image for a Python application."
```
Comment (MCazaly) [inline java-demo/Dockerfile:16]: This probably needs a small correction?
```suggestion
LABEL org.opencontainers.image.description="Example OCI image for a Java application."
```

**Review by MCazaly — CHANGES_REQUESTED** [review]
Looking good. Just two tiny tweaks left for our labels.

**Comment by MCazaly** [comment]
> @MCazaly also added some tools and docs to run the tagged image from the registry (not simple as it requires a database). Actually you need to clone the repository to get docker compose files, I'll certainly have to improve that in the future.

That's fine - the compose files are actually a good example for how images might be used, so it's a good thing!

**Comment by leobr29** [comment]
@MCazaly I think you need to validate changes so I can merge

**Pull request merged**